rq/job.py

```diff
@@ -393,6 +398,32 @@ def dependents_key(self):
         """The Redis key that is used to store job dependents hash under."""
         return self.dependents_key_for(self.id)
 
+    @property
+    def dependencies_key(self):
+        return '{0}:{1}:dependencies'.format(self.redis_job_namespace_prefix, self.id)
+
+    def fetch_dependencies(self, watch=False, pipeline=None):
+        """
+        Fetch all of a job's dependencies. If a pipeline is supplied, and
+        watch is true, then set WATCH on all the keys of all dependencies.
+
+        Returned jobs will use self's connection, not the pipeline supplied.
+        """
+        if pipeline is None:
+            pipeline = pipeline or self.connection
+
+        if watch and self._dependency_ids:
+            pipeline.watch(*self._dependency_ids)
+
+        jobs = self.fetch_many(self._dependency_ids, connection=self.connection)
+
+        for i, job in enumerate(jobs):
+            if not job:
+                raise InvalidJobDependency('Job {0} does not exist'.format(self._dependency_ids[i]))
```

Comment:
One last comment, I'm just wondering whether we should raise a `NoSuchJobError('Dependency {0} does not exist')` instead of `InvalidJobDependency`. I could go either way. Thoughts?

Reply:
I think `NoSuchJobError` is the way to go. It seems a bit more consistent with what is raised other places in `job.py` and also seems to more precisely describe _what went wrong_.

Just to call out: In the environments in which _I_ use RQ in production, this problem will surface in logs and alerting - raising an exception named `InvalidJobDependency` as opposed to `NoSuchJobError` isn't going to provide any context I won't already have from a stack trace. To that end, I only see any place _else_ `InvalidJobDependency` is used. Perhaps it could be removed?

Reply:
Ok.